compass/landice/tests/ismip6_forcing/atmosphere/process_smb.py

```diff
@@ -0,0 +1,340 @@
+        """
+
+        # check if a mapfile exists
+        mapping_file = f"map_ismip6_8km_to_{mali_mesh_name}_{method_remap}.nc"
+
+        if not os.path.exists(mapping_file):
+            # build a mapping file if it doesn't already exist
+            build_mapping_file(input_file, mapping_file, mali_mesh_file,
+                               method_remap)
+        else:
+            self.logger.info("Mapping file exists. "
+                             "Remapping the input data...")
+
+        # remap the input data
+        args = ["ncremap",
+                "-i", input_file,
+                "-o", output_file,
+                "-m", mapping_file,
+                "-v", "smb_anomaly"]
+
+        subprocess.check_call(args)
+
+    def rename_ismip6smb_to_mali_vars(self, remapped_file_temp, output_file):
+        """
+        Rename variables in the remapped ismip6 input data
+        to the ones that MALI uses.
+
+        Parameters
+        ----------
+        remapped_file_temp : str
+            temporary ismip6 data remapped on mali mesh
+        output_file : str
+            remapped ismip6 data renamed on mali mesh
+        """
+
+        # open dataset in 20 years chunk
+        ds = xr.open_dataset(remapped_file_temp, chunks=dict(time=20),
+                             engine="netcdf4")
+
+        # build dictionary for ismip6 variables that MALI takes in
+        ismip6_to_mali_dims = dict(
+            time="Time",
+            ncol="nCells")
+        ds = ds.rename(ismip6_to_mali_dims)
+
+        ismip6_to_mali_vars = dict(
+            smb_anomaly="sfcMassBal")
+        ds = ds.rename(ismip6_to_mali_vars)
+
+        # add xtime variable
+        xtime = []
+        for t_index in range(ds.sizes["Time"]):
+            date = ds.Time[t_index]
+            date = date.dt.strftime("%Y-%m-%d_00:00:00")
+            date = str(date.values).ljust(64)
+            xtime.append(date)
+
+        ds["xtime"] = ("Time", xtime)
+        ds["xtime"] = ds.xtime.astype('S')
+
+        # drop unnecessary variables
+        ds = ds.drop_vars(["lon", "lon_vertices", "lat", "lat_vertices",
+                           "area"])
+
+        # write to a new netCDF file
+        write_netcdf(ds, output_file)
+        ds.close()
+
+    def correct_SMB_anomaly_for_baseSMB(self, output_file, mali_mesh_file):
+        """
+        Apply the MALI base SMB to the ismip6 SMB anomaly field
+
+        Parameters
+        ----------
+        output_file : str
+            remapped ismip6 data renamed on mali mesh
+        mali_mesh_file : str
+            initialized MALI mesh file in which the base SMB field exists
+        """
+
+        ds = xr.open_dataset(output_file)
+        ds_base = xr.open_dataset(mali_mesh_file)
+        # get the first time index
+        ref_smb = ds_base["sfcMassBal"].isel(Time=0)
+        # broadcast so time is the same size as in ds
+        ref_smb = ref_smb.broadcast(ds["sfcMassBal"])
```

Comment:
This error is a weird one because I ran into it once, but then it disappeared at another time, which is why I committed the changes. This was something that I think @xylar was also confused about and thought there was nothing wrong with the line of the code itself and thus shouldn't give an error. Good news is that this line is not really necessary and the code runs as expected without it. I've taken the line out, and the testcase should work all fine now.

Reply:
Turns out the syntax we were using were not quite correct. Still puzzling though why I thought it worked for me. I must have tested in a wrong branch that didn't have the line `ref_smb = ref_smb.broadcast(ds["sfcMassBal"])`. Anyhow, we still don't need the line as it is functionally redundant.